functions.php

```diff
@@ -56,3 +55,23 @@ function displayCard(array $cards) : string
     }
     return $result;
 }
+
+function allDataPresent(array $userData): bool
+{
+    return (isset($userData['name']) && isset($userData['type_line']) && isset($userData['mana']) && isset($userData['rarity']) && isset($userData['img_url']));
+}
+
+// validation
+function validLength(string $data): bool
```

Comment:
Just a suggestion, may want to add in something to not allow special characters as well as checking length.

Reply:
Nicely noted. Just not sure if I'll have the time :(